src/Extensions/MaintenanceLockExtension.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Symbiote\QueuedJobs\Extensions;
+
+use SilverStripe\Forms\CheckboxField;
+use SilverStripe\Forms\FieldList;
+use SilverStripe\ORM\DataExtension;
+use SilverStripe\SiteConfig\SiteConfig;
+use Symbiote\QueuedJobs\Services\QueuedJobService;
+
+/**
+ * Class MaintenanceLockExtension
+ * Adds a maintenance lock UI to SiteConfig
+ *
+ * @property SiteConfig|$this owner
+ * @package Symbiote\QueuedJobs\Extensions
+ */
+class MaintenanceLockExtension extends DataExtension
+{
+    /**
+     * @param FieldList $fields
+     */
+    public function updateCMSFields(FieldList $fields)
+    {
+        if (!QueuedJobService::config()->get('lock_file_enabled')) {
+            return;
+        }
+
+        $fields->addFieldsToTab('Root.QueueSettings', [
+            $lockField = CheckboxField::create(
+                'MaintenanceLockEnabled',
+                'Maintenance Lock Enabled',
+                QueuedJobService::isMaintenanceLockActive()
```

Comment:
This might be personal preference, but I think this should be a non-static method on the `QueuedJobService` singleton

Reply:
I think this suggestion is functionally nicer as you're removing global state. With your suggestion you will be able to (theoretically) have two `QueuedJobService`s with different "maintenance lock" state.